jetbrains-core/src/software/aws/toolkits/jetbrains/core/plugins/ToolkitUpdateManager.kt

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.plugins
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor
+import com.intellij.ide.plugins.PluginManagerCore
+import com.intellij.notification.NotificationAction
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.service
+import com.intellij.openapi.extensions.PluginId
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.updateSettings.impl.PluginDownloader
+import com.intellij.openapi.updateSettings.impl.UpdateChecker
+import com.intellij.util.Alarm
+import com.intellij.util.concurrency.annotations.RequiresBackgroundThread
+import software.aws.toolkits.core.utils.debug
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.jetbrains.AwsToolkit
+import software.aws.toolkits.jetbrains.settings.AwsSettings
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+import software.aws.toolkits.resources.message
+
+class ToolkitUpdateManager {
+    private val alarm = Alarm(Alarm.ThreadToUse.SWING_THREAD)
+
+    init {
+        runTask()
+    }
+
+    private fun runTask() {
+        if (alarm.isDisposed) return
+        scheduleUpdateTask()
+
+        if (!AwsSettings.getInstance().isAutoUpdateEnabled) return
+
+        ProgressManager.getInstance().run(
+            object : Task.Backgroundable(null, "Updating AWS Toolkit") {
+                override fun run(indicator: ProgressIndicator) {
+                    checkForUpdates(indicator)
+                }
+            }
+        )
+    }
+
+    private fun scheduleUpdateTask() {
+        alarm.addRequest({ runTask() }, UPDATE_CHECK_INTERVAL_IN_MS)
+    }
+
+    @RequiresBackgroundThread
+    fun checkForUpdates(progressIndicator: ProgressIndicator) {
+        // Note: This will need to handle exceptions and ensure thread-safety
+        try {
+            val toolkitPlugin = PluginManagerCore.getPlugin(PluginId.findId(AwsToolkit.PLUGIN_ID)) ?: return
+            if (!toolkitPlugin.isEnabled) return
+            val latestToolkitPluginDownloader = getUpdate(toolkitPlugin) ?: return
+            if (!latestToolkitPluginDownloader.prepareToInstall(progressIndicator)) return
+            latestToolkitPluginDownloader.install()
+        } catch (e: Exception) {
+            LOG.debug(e) { "Unable to install AWS Toolkit" }
+            return
+        }
+        if (AwsSettings.getInstance().isAutoUpdateNotificationDoNotShowAgain) return
```

Comment:
we have a checkbox in the AWS settings page and we repsect this setting by 
 `if (!AwsSettings.getInstance().isAutoUpdateEnabled) return`

Reply:
Should we also send a first time notification saying we have this new feature where it will download updates automatically? We can provide them an option to disable it in this notification, similar to what we do for telemetry

Reply:
It can be a good idea, will wait for the feedback and implement that once that comes up.